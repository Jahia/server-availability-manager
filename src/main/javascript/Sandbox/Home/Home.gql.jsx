import gql from 'graphql-tag';

const getCurrentUserName = gql`
    {
      currentUser {
        name
      }
    }
`;

export {getCurrentUserName};
