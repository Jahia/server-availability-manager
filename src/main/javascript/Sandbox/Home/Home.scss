.root {
    display: flex;
    flex-direction: column;
    min-height: 100%;
    padding: 0;

    background-color: var(--color-gray_light40);
    box-shadow: 0 4px 4px rgba(59, 61, 64, 0.3);
}

.headerRoot {
    padding: var(--spacing-medium) var(--spacing-large) 0 var(--spacing-large);

    background-color: var(--color-white);
    box-shadow: 0 4px 8px var(--color-dark40);
}

.header {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-start;
    min-height: 48px;
}

.content {
    padding: var(--spacing-medium) var(--spacing-large) var(--spacing-medium) var(--spacing-large);

    background-color: var(--color-gray_light40);
}

.actionBar {
    display: flex;
    flex-grow: 0.05;
    flex-direction: row-reverse;
    justify-content: space-between;
    align-items: center;
}
