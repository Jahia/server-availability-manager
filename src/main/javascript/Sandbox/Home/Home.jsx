import React, {useState} from 'react';
import {useTranslation} from 'react-i18next';
import {Add, Button, GlobalStyle, Typography} from '@jahia/moonstone';
import styles from './Home.scss';
import {getCurrentUserName} from './Home.gql';
import {useQuery} from '@apollo/react-hooks';

const Home = () => {
    const {t} = useTranslation('sandbox');
    const [userInformation, setUserInformation] = useState({username: ''});

    const updateCurrentUser = data => {
        setUserInformation({...userInformation, username: data.currentUser.name});
    };

    useQuery(getCurrentUserName, {
        onCompleted: updateCurrentUser
    });

    return (
        <React.Fragment>
            <GlobalStyle/>
            <div className={styles.root}>
                <div className={styles.headerRoot}>
                    <header className={styles.header}>
                        <Typography variant="title">{t('sandbox:title')}
                        </Typography>
                        <div className={styles.actionBar}>
                            <Button size="big"
                                    color="accent"
                                    label={t('sandbox:action.buttonTitle')}
                                    icon={<Add/>}
                                    onClick={() => {}}/>
                        </div>
                    </header>
                </div>
                <div className={styles.content}>
                    <span>Some content, connected with: {userInformation.username}</span>
                </div>
            </div>
        </React.Fragment>
    );
};

export default Home;
