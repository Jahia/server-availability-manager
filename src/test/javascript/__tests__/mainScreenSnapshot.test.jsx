import React from 'react';
import {act, cleanup, render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Home from '../../../main/javascript/Sandbox/Home/Home';
import {sandboxMocks} from '../apolloMocks';
import {MockedProvider, wait} from '@apollo/react-testing';

describe('Validate main screen snapshot', () => {
    afterEach(() => {
        cleanup();
    });
    test('match snapshot', async () => {
        const {container} = render(
            <MockedProvider mocks={sandboxMocks} addTypename={false}>
                <Home/>
            </MockedProvider>
        );
        await act(async () => {
            await wait(0);
        });
        expect(container).toMatchSnapshot();
    });
});

