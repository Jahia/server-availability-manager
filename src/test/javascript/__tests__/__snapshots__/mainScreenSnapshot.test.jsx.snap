// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate main screen snapshot match snapshot 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="headerRoot"
    >
      <header
        class="header"
      >
        <p
          class="moonstone-typography moonstone-variant_title moonstone-weight_default"
        >
          translated_sandbox:title
        </p>
        <div
          class="actionBar"
        >
          <button
            class="moonstone-button moonstone-size_big moonstone-variant_default moonstone-color_accent moonstone-icon"
            type="button"
          >
            <svg
              class=" moonstone-icon moonstone-icon_default"
              fill="none"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                fill="currentColor"
              />
            </svg>
            <span
              class="moonstone-typography moonstone-variant_button moonstone-weight_semiBold flexFluid moonstone-nowrap moonstone-upperCase"
            >
              translated_sandbox:action.buttonTitle
            </span>
          </button>
        </div>
      </header>
    </div>
    <div
      class="content"
    >
      <span>
        Some content, connected with: 
        JohnDoe
      </span>
    </div>
  </div>
</div>
`;
